vermouth/pdb/pdb.py

```diff
@@ -16,13 +16,408 @@
+    exclude: collections.abc.Container[str]
+        Container of residue names. Any atom that has a residue name that is in
+        `exclude` will be skipped.
+    ignh: bool
+        Whether all hydrogen atoms should be skipped
+    modelidx: int
+        Which model to take.
+    """
+
+    def __init__(self, exclude=('SOL',), ignh=False, modelidx=1):
+        self.active_molecule = Molecule()
+        self.molecules = []
+        self._conects = []
+        self.exclude = exclude
+        self.ignh = ignh
+        self.modelidx = modelidx
+        self._skipahead = False
+
+    def dispatch(self, line):
+        """
+        Returns the appropriate method for parsing `line`. This is determined
+        based on the first 6 characters of `line`.
+
+        Parameters
+        ----------
+        line: str
+
+        Returns
+        -------
+        collections.abc.Callable[str, int]
+            The method to call with the line, and the line number.
+        """
+        record = line[:6].strip().lower()
+        return getattr(self, record, self._unknown_line)
+
+    def parse(self, file_handle):
+        # Only PDBParser.finalize should produce a result, namely a list of
+        # molecules. This means that mols is a list containing a single list of
+        # molecules, which is a little silly.
+        outcome = list(super().parse(file_handle))
+        assert len(outcome) == 1
+        yield from outcome[0]
+
+    @staticmethod
+    def _unknown_line(line, lineno):
+        """
+        Called when a line is unknown. Raises a KeyError with a helpful message.
+        """
+        raise KeyError("Line {} can not be parsed, since we don't recognize the"
+                       " first 6 characters. The line is: {}"
+                       "".format(lineno, line))
+
+    @staticmethod
+    def _skip(line, lineno=0):
+        """
+        Does nothing.
+        """
+
+
+    # TODO: Parse some of these, and either do something useful with it, or
+    #       propagate it to e.g. the ITP
+    # pylint: disable=bad-whitespace
+    # TITLE SECTION
+    header = _skip
+    obslte = _skip
+    title  = _skip
+    splt   = _skip
+    caveat = _skip
+    compnd = _skip
+    source = _skip
+    keywds = _skip
+    expdta = _skip
+    nummdl = _skip
+    mdltyp = _skip
+    author = _skip
+    revdat = _skip
+    sprsde = _skip
+    jrnl   = _skip
+    remark = _skip
+
+    # PRIMARY STRUCTURE SECTION
+    dbref  = _skip
+    dbref1 = _skip
+    dbref2 = _skip
+    seqadv = _skip
+    seqres = _skip
+    modres = _skip
+
+    # HETEROGEN SECTION
+    het    = _skip
+    formul = _skip
+    hetnam = _skip
+    hetsyn = _skip
+
+    # SECONDARY STRUCTURE SECTION
+    helix  = _skip
+    sheet  = _skip
+
+    # CONNECTIVITY ANNOTATION SECTION
+    ssbond = _skip
+    link   = _skip
+    cispep = _skip
+
+    # MISCELLANEOUS FEATURES SECTION
+    site   = _skip
+
+    # CRYSTALLOGRAPHIC AND COORDINATE TRANSFORMATION SECTION
+    cryst1 = _skip
```

Comment:
This is the unit cell and should probably be retained at some point to allow coarse graining a full simulation system. A comment will do here for now and an issue for reference.